TYPO3.TypoScript/Classes/TYPO3/TypoScript/Core/Cache/RuntimeContentCache.php

```diff
@@ -175,25 +175,11 @@ public function preEvaluate(array &$evaluateContext, $tsObject)
                         $result = $self->evaluateUncached($path, $unserializedContext);
                         return $result;
                     } elseif (strpos($command, 'evalCached=') === 0) {
-                        $identifier = substr($command, 11);
-                        $cacheDiscriminator = $this->runtime->evaluate($additionalData['path'] . '/__meta/cache/entryDiscriminator');
-                        if ($cacheDiscriminator === false) {
-                            $unserializedContext = $self->unserializeContext($additionalData['context']);
-                            return $self->evaluateUncached($additionalData['path'], $unserializedContext);
-                        }
-                        $cacheIdentifier = substr($identifier, 0, strpos($identifier, '_')) . '_' . md5($cacheDiscriminator);
-                        $result = $cache->get($cacheIdentifier);
-                        if ($result === false) {
-                            $unserializedContext = $self->unserializeContext($additionalData['context']);
-                            $maximumLifetime = null;
-                            if (isset($evaluateContext['configuration']['maximumLifetime'])) {
-                                $maximumLifetime = $this->runtime->evaluate($evaluateContext['typoScriptPath'] . '/__meta/cache/maximumLifetime', $tsObject);
-                            }
-                            $cacheTags = $this->buildCacheTags($evaluateContext['configuration'], $evaluateContext['typoScriptPath'], $tsObject);
-                            $result = $self->evaluateUncached($additionalData['path'], $unserializedContext);
-                            $cache->set($cacheIdentifier, $result, $cacheTags, $maximumLifetime);
-                        }
-
+                        $self->inCacheEntryPoint = null;
```

Comment:
I wonder if this is (always) correct here?

Is this basically a trick to not create an uncached segment in `evaluate`?

Reply:
I fear that playing with that state could have side effects for other cache segments on the same level, so e.g. we might need to reset it to the original value after calling `evaluate`.

Reply:
It's a trick to make it always process cache segments at this point, otherwise it would keep them if it was set in the parent. The parent state will still be correct as at this point it is already in the cacheContext as true/false. This property only prevents processing in nested entries, which in this case is required. Technically we should remove this property and always look into the current cache context if it's a cache entryPoint or not, but that then has other difficulties I guess ;)

Reply:
To add to the above, the closure is only for replacing nested segments, so it's not expected that the result contains cache segment markers, it would though if we run through evaluate, so we must make sure they get removed again. As we KNOW that this closure is only used for cached segments and those are always cache entry points there cannot be any other segments on the same level that would need the property value as either they were already cached and so it doesn't matter OR they need to be re-rendered at which point the whole parent element is re-rendered anyway (mmm and thinking about it, this could be the point where it goes wrong.... Needs to be tested).